core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -785,6 +782,11 @@ public void terminateWorkflow(Workflow workflow, String reason, String failureWo
             if (workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
                 workflowStatusListener.onWorkflowTerminated(workflow);
             }
+
+            //remove from the sweep queue
```

Comment:
tasks should be removed from queue before status is updated, else tasks to be CANCELED will be polled and processed by the workers.